From 3e2b2c2e12c89d9de1e25cfc107bdfe01e010e4c Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Sun, 09 Aug 2020 08:08:56 +0300
Subject: [PATCH] [SW-13299] hl: i2c-slave: remove interrupt mask

Remove interrupt mask in i2c-slave initialization.
The i2c-slave device can operate in 2 modes:
Operational mode (in which device is configured as address 0x4f)
and debug mode (in which device is configured as address 0x47).
On reset, it boots with default interruot mask, but then in previous
patch in preboot, the interrupt mask was modified to allow all
interrupts. It sometimes resulted in non responsive device.
So, in this patch we are using the default mask interrupt for
the i2c-slave device (both in debug and in operational modes).

Change-Id: Iffc396f64ae7221660af5b6e848b07b1e00c92b4
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index 6ce1ee5..9ba3c9c 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -49,10 +49,6 @@
 	/* Program IC_CON */
 	regmap_writel(0x22, mmPSOC_I2C_S_IC_CON);
 
-	/* Write to IC_INTR_MASK to enable all interrupts */
-	if (!debug_mode)
-		regmap_writel(0x7FFF, mmPSOC_I2C_S_IC_INTR_MASK);
-
 	/* Write to IC_RX_TL to set Rx FIFO threshold level */
 	regmap_writel(0x0, mmPSOC_I2C_S_IC_RX_TL);
 
